Source/Core/VideoBackends/Software/Tev.cpp

```diff
@@ -369,25 +369,34 @@ static bool TevAlphaTest(int alpha)
 
 static inline s32 WrapIndirectCoord(s32 coord, int wrapMode)
 {
+	s32 size;
+
 	switch (wrapMode)
 	{
-		case ITW_OFF:
-			return coord;
 		case ITW_256:
-			return (coord % (256 << 7));
+			size = (256 << 7);
+			break;
 		case ITW_128:
-			return (coord % (128 << 7));
+			size = (128 << 7);
+			break;
 		case ITW_64:
-			return (coord % (64 << 7));
+			size = (64 << 7);
+			break;
 		case ITW_32:
-			return (coord % (32 << 7));
+			size = (32 << 7);
+			break;
 		case ITW_16:
-			return (coord % (16 << 7));
+			size = (16 << 7);
+			break;
 		case ITW_0:
 			return 0;
+		case ITW_OFF:
+			return coord;
 		default:
 			return 0;
 	}
+
+	return (coord < 0) ? (size - (std::abs(coord) % size)) : (std::abs(coord) % size);
```

Comment:
Isn't this ternary equivalent to:
```C++
return (coord < 0) ? (size + coord % size) : (coord % size);
```

Which can be written as such:
```C++
return (size + coord % size) % size;
```

No need to use the absolute value when we know it's negative or not. I'm guessing the size can't be zero, otherwise it'd be really dangerous. The signess of the size doesn't seem to affect the result sign on gcc (I don't know for the others compilers).